simnode/generator.py

```diff
@@ -108,27 +233,35 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
 
                 # Draw line separating left from right lanes
                 if road.left_lanes > 0 and road.right_lanes > 0:
-                    offset = initial_line_offsets[road.left_lanes]
+                    divider_rel_off = -0.5 + road.left_lanes/(road.left_lanes+road.right_lanes)
+                    offset_list_divider = _calculate_offset_list(relative_offset=divider_rel_off,
+                                                                 absolute_offset=-side_line_offset)
                     left_right_divider = Road("line_yellow_double", rid=road.rid + "_left_right_divider")
-                    left_right_divider_nodes \
-                        = _calculate_parallel_coords(offset, 2 * self.add_roads_to_scenario.line_width)
+                    left_right_divider_nodes = \
+                        _calculate_parallel_coords_list(offset_list_divider, 2 * self.add_roads_to_scenario.line_width)
                     if left_right_divider_nodes:
                         left_right_divider.nodes.extend(left_right_divider_nodes)
                         scenario.add_road(left_right_divider)
                     else:
                         _logger.warning("Could not create line separating lanes having different directions")
 
                 # Draw lines separating left and right lanes from each other
-                offset_indices = []
-                if road.left_lanes > 1:
-                    offset_indices.extend(range(1, road.left_lanes))
-                if road.right_lanes > 1:
-                    offset_indices.extend(range(road.left_lanes + 1, len(initial_line_offsets) - 1))
-                for index in offset_indices:
-                    offset = initial_line_offsets[index]
-                    lane_separation_line = Road('line_dashed_short', rid=road.rid + "_separator_" + str(index))
+                total_num_lane_markings = road.left_lanes + road.right_lanes
+                offsets_dashed = linspace(-0.5, 0.5, num=total_num_lane_markings, endpoint=False)
+                offsets_dashed = offsets_dashed[1:len(offsets_dashed)]  # why does 1:-1 not work?
+                for offs in offsets_dashed:
+                    # do not draw dashed line over divider line
+                    if (road.left_lanes > 1 and road.right_lanes > 0) or (road.left_lanes > 0 and road.right_lanes > 1):
+                        if offsets_dashed[road.left_lanes - 1] + 0.01 > offs > offsets_dashed[road.left_lanes - 1] - 0.01:
```

Comment:
Concerning this condition some stuff including "filter" and "map" might clarify the structure.